### ISSUE TITLE
[JENKINS-8950] CLONE - Disable polling if build is running

### ISSUE THREAD

**Issue by timja-bot** [timja-org/jenkins-gh-issues-poc-11-04#7650]
<p><font color="red">DUPLICATE: <a href="http://issues.hudson-ci.org/browse/HUDSON-8674*" class="external-link" target="_blank" rel="nofollow noopener">http://issues.hudson-ci.org/browse/HUDSON-8674*</a></font></p>

<p>We enabled minutely SCM polling. If build is running (about 2 hours) we see lots of polling executions (lshistory). This slow down the proccess. It should be possible to disable polling if build is running.</p>

---
<details><summary><i>Originally reported by <a title="anb0s" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=anb0s">anb0s</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-8950" target="_blank">CLONE - Disable polling if build is running</a></i></summary>
<i><ul>
<li><b>status</b>: Reopened
<li><b>priority</b>: Major
<li><b>resolution</b>: Unresolved
<li><b>imported</b>: 2025-11-05
</ul></i>
</details>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Nobody is interested in this feature, so resolving as Won't Fix.</p>

<p>Also, it's not obvious why a new build shouldn't be queued after the current build checked out from SCM and a new change was detected.</p>

<p>Querying SCM every minute is unnecessarily expensive anyway, if you need immediate builds, just switch to post-commit hook notifications.</p>

<p>Also, behavior would need to be determined for parallel build projects.</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=anb0s">anb0s</a>:</i>
<p>I'm interested and this trivial patch works for us with ClearCase for years. We had no option for hooks at server side and ClearCase had no support for atomic commits <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/> But maybe it's possible with ClearCase 8 and CMs are now open for new ideas <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/wink.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>Anyway if hooks are not possible with some SCMs than this is the only way to get SCM plugin with polling mechanism working fast enought and without starting builds in parallel. Why should the system still poll for changes if build is running? It should poll after the build was finished and then build the next changes in one time...</p>
